core/src/main/java/apoc/util/JsonUtil.java

```diff
@@ -115,19 +119,49 @@ public static <T> T parse(String json, String path, Class<T> type) {
     }
     
     public static <T> T parse(String json, String path, Class<T> type, List<String> options) {
+        return parse(json, path, type, options, true, false);
+    }
+    
+    public static <T> T parse(String json, String path, Class<T> type, List<String> options, JsonConfig config) {
+        return parse(json, path, type, options, config.isFailOnError(), false);
+    }
+    
+    public static <T> T parse(String json, String path, Class<T> type, List<String> options, boolean failOnError, boolean validation) {
+        
         if (json==null || json.isEmpty()) return null;
         try {
+            SilentDeserializer deserializer = new SilentDeserializer(validation, null, null);
+            ObjectMapper objectMapper = getObjectMapper(deserializer);
             final String listOpt = Option.ALWAYS_RETURN_LIST.name();
             if (type == Map.class && options != null && options.contains(listOpt)) {
                 throw new RuntimeException("It's not possible to use " + listOpt + " option because the conversion should return a Map");
             }
             if (path == null || path.isEmpty()) {
-                return OBJECT_MAPPER.readValue(json, type);
+                final T t = (T) objectMapper.readValue(json, Object.class);
+                return getJson(deserializer, t, validation);
             }
```

Comment:
Unfortunately, I believe that deserializing any json to an Object is a fairly Jackson vulnerability. I found [this article](https://security.snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1056418) about it which could explain it better than I could. 

The idea is that an attacker can make you build any class on your classpath that they like. And one of those classes on your class path might take java as a string in their constructor method, and run that java essentially.

If I'm correct and this is vulnerable, then I think there are several solutions that could be considered from the article. I would leave the implementation up to you. But I think that we ought to do what's possible to not introduce this vulnerability if possible.